shim/chaincodeserver.go

```diff
@@ -0,0 +1,79 @@
+// Copyright the Hyperledger Fabric contributors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package shim
+
+import (
+	"crypto/tls"
+	"errors"
+
+	"github.com/hyperledger/fabric-chaincode-go/shim/internal"
+	pb "github.com/hyperledger/fabric-protos-go/peer"
+
+	"google.golang.org/grpc/keepalive"
+)
+
+// TLSProperties passed to ChaincodeServer
+type TLSProperties struct {
+	//Disabled forces default to be TLS enabled
+	Disabled bool
+	Key      []byte
+	Cert     []byte
+	// ClientCACertsPath set if connecting peer should be verified
+	ClientCACerts []byte
+}
+
+// ChaincodeServer encapsulates basic properties needed for a chaincode server
+type ChaincodeServer struct {
+	// Name should match chaincode's package name on peer
+	Name string
+	// Addesss should match chaincode's address in package on peer
+	Address string
+	// CC is the chaincode that handles Init and Invoke
+	CC Chaincode
+	// TLSProps
```

Comment:
I'd either fill in or remove this comment.

Reply:
Will fill.